### PR TITLE
Add an option to display all dates in UTC

### PR DESCRIPTION
**Pull request by ngorugan**
Added an option -Z to display all dates in UTC.

Resolves ipmitool/ipmitool#19
Change-Id: Iac3a61190eefde12d95c892af26072ec01f60474
Signed-off-by: Nagaraju Goruganti <ngorugan@in.ibm.com>

### CODE REVIEW

File: lib/ipmi_chassis.c [hunk 1]
```diff
@@ -716,11 +716,22 @@ ipmi_chassis_get_bootparam(struct ipmi_intf * intf, char * arg)
...
+				);
+			}
+			else
+			{
+				strftime(
+						time_buf,
+						sizeof(time_buf),
+						"%m/%d/%Y %H:%M:%S", localtime(&out_time)
+				);
+			}
```
[the remark this answers is not in the document]
Comment (ngorugan) [reply lib/ipmi_chassis.c]: Done

File: lib/ipmi_ekanalyzer.c [hunk 2]
```diff
@@ -2546,8 +2546,12 @@ ipmi_ek_display_fru_header_detail(char *filename)
...
 		tval = tval * 60;
 		tval = tval + secs_from_1970_1996;
-		printf("Board Mfg Date: %ld, %s", tval,
-				asctime(localtime(&tval)));
+		if(time_in_utc)
+			printf("Board Mfg Date: %ld, %s", tval,
+					asctime(gmtime(&tval)));
+		else
+			printf("Board Mfg Date: %ld, %s", tval,
+					asctime(localtime(&tval)));
```
[the remark this answers is not in the document]
Comment (ngorugan) [reply lib/ipmi_ekanalyzer.c]: Done

File: lib/ipmi_fru.c [hunk 3]
```diff
@@ -1034,7 +1034,10 @@ fru_area_print_board(struct ipmi_intf * intf, struct fru_info * fru,
 	tval=((fru_data[i+2] << 16) + (fru_data[i+1] << 8) + (fru_data[i]));
 	tval=tval * 60;
 	tval=tval + secs_from_1970_1996;
-	printf(" Board Mfg Date        : %s", asctime(localtime(&tval)));
+	if(time_in_utc)
+		printf(" Board Mfg Date        : %s", asctime(gmtime(&tval)));
+	else
+		printf(" Board Mfg Date        : %s", asctime(localtime(&tval)));
```
[the remark this answers is not in the document]
Comment (ngorugan) [reply lib/ipmi_fru.c]: Done

File: lib/ipmi_main.c [hunk 4]
```diff
@@ -263,6 +264,7 @@ ipmi_option_usage(const char * progname, struct ipmi_cmd * cmdlist, struct ipmi_
 	lprintf(LOG_NOTICE, "       -O seloem      Use file for OEM SEL event descriptions");
 	lprintf(LOG_NOTICE, "       -N seconds     Specify timeout for lan [default=2] / lanplus [default=1] interface");
 	lprintf(LOG_NOTICE, "       -R retry       Set the number of retries for lan/lanplus interface [default=4]");
+	lprintf(LOG_NOTICE, "       -Z timezone    Display all dates in UTC");
```
[the remark this answers is not in the document]
Comment (ngorugan) [reply lib/ipmi_main.c]: Done

File: lib/ipmi_mc.c [hunk 5]
```diff
@@ -539,7 +539,10 @@ ipmi_mc_print_guid(struct ipmi_guid_t guid)
 	       guid.node[3], guid.node[4], guid.node[5]);
 
 	s = (time_t)guid.time_low; /* Kipp - removed the BSWAP_32, it was not needed here */
-	strftime(tbuf, sizeof(tbuf), "%m/%d/%Y %H:%M:%S", localtime(&s));
+	if(time_in_utc)
+		strftime(tbuf, sizeof(tbuf), "%m/%d/%Y %H:%M:%S", gmtime(&s));
+	else
+		strftime(tbuf, sizeof(tbuf), "%m/%d/%Y %H:%M:%S", localtime(&s));
```
[the remark this answers is not in the document]
Comment (ngorugan) [reply lib/ipmi_mc.c]: Done

File: lib/ipmi_sel.c [hunk 6]
```diff
@@ -52,6 +52,7 @@
 #include <ipmitool/ipmi_strings.h>
 
 extern int verbose;
+extern int time_in_utc;
```
[the remark this answers is not in the document]
Comment (ngorugan) [reply lib/ipmi_sel.c]: Done

File: lib/ipmi_main.c [hunk 7]
```diff
@@ -92,6 +92,7 @@ ipmi_intf_set_max_request_data_size(struct ipmi_intf * intf, uint16_t size);
 
 extern int verbose;
 extern int csv_output;
+extern int time_in_utc;
```
[the remark this answers is not in the document]
Comment (ngorugan) [reply lib/ipmi_main.c]: Done

**Comment by AlexanderAmelkin** [comment]
Ok, it's fine now except that it doesn't "display **all** times in UTC" when `-Z` is specified, or to be more correct, not **all** times are displayed in local time whe `-Z` is NOT specified.

For instance, there are still places `ipmi_sdr.c`, `ipmi_sel.c`, and other places where times are converted using `gmtime()` regardless of `-Z`. You can `grep -Erl --include '*.c' 'strftime|asctime'` to get the list of files.

If we're adding an option that is said to force UTC time zone, then users will expect local time when the option is not specified. Let's make all times displayed in the same time zone everywhere in the tool.

And it just came to my mind that if we want to universally convert `time_t` timestamps to strings (and there are lots of places where we do that), then we need a universal single function for that. Somewhere in helper.c, I guess. Moreover, there already is such a function, as it turns out. In `ipmi_delloem.c` they have `ipmi_time_to_str()`. And another one is `ipmi_sdr_timestamp()`. Maybe there are even more of them.

But I guess that's another separate effort.

**Pull request merged**